File: bridge/src/bridge_daily_rewards.cpp
#if defined(DM_PLATFORM_HTML5)
#include "bridge_daily_rewards.h"
#include "bridge.h"
#include "bridge_helper.h"

int bridge::dailyRewards::getRewards(lua_State* L) {
    return makeCallback(L, js_bridge_daily_rewards_getRewards, true);
}

int bridge::dailyRewards::getCurrentDay(lua_State* L) {
    return makeCallback(L, js_bridge_daily_rewards_getCurrentDay, true);
}

int bridge::dailyRewards::getCurrentReward(lua_State* L) {
    return makeCallback(L, js_bridge_daily_rewards_getCurrentReward, true);
}

int bridge::dailyRewards::claimCurrentReward(lua_State* L) {
    return makeCallback(L, js_bridge_daily_rewards_claimCurrentReward, true);
}

#endif

File: bridge/bridge_daily_rewards.lua
local daily_rewards = {}

function daily_rewards.get_rewards(on_success, on_failure)
	if on_failure then
		on_failure()
	end
end

function daily_rewards.get_current_day(on_success, on_failure)
	if on_failure then
		on_failure()
	end
end

function daily_rewards.get_current_reward(on_success, on_failure)
	if on_failure then
		on_failure()
	end
end

function daily_rewards.claim_current_reward(on_success, on_failure)
	if on_failure then
		on_failure()
	end
end

return daily_rewards
